nixos/modules/services/games/factorio.nix

```diff
@@ -283,6 +283,23 @@ in
           Autosaving on connected Windows clients will be disabled regardless of autosave_only_on_server option.
         '';
       };
+      rConPassword = lib.mkOption {
+        type = lib.types.nullOr lib.types.str;
+        default = null;
+        description = ''
+          Sets the rcon server password.
```

Comment:
Probably remind the user that this will be world-readable in the store. Perhaps consider if we want to provide a from-file option as well or instead?

Reply:
I'll take a stab at those suggestions, I haven't made a module with a submodule like that before but I think it makes sense in this case since both arguments must be present together.

Reply:
thanks; if you find it tricky I'm happy to try to come up with an implementation tomorrow or so; I haven't used submodules before either but I'm pretty sure I've seen examples around

Reply:
ah, and in fact even if something like that `$(cat ...)` worked it would be a pretty weak improvement since your process command line is visible in `/proc`...

Sounds like we can't really make this secure from other users on the same host unless Factorio adds the ability to read the password from a file. Sorry for leading you down a doomed path :)

Reply:
see also: https://forums.factorio.com/viewtopic.php?f=6&t=51325&p=413867

(I decided not to ping it since I don't personally use rcon anyway)